middleware/basicauth/basicauth.go

```diff
@@ -64,6 +73,74 @@ func (a BasicAuth) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error
 // file or directory paths.
 type Rule struct {
 	Username  string
-	Password  string
+	Password  func(string) bool
 	Resources []string
 }
+
+type PasswordMatcher func(pw string) bool
+
+var (
+	htpasswords   map[string]map[string]PasswordMatcher
+	htpasswordsMu sync.Mutex
+)
+
+func GetHtpasswdMatcher(filename, username string) (PasswordMatcher, error) {
+	filename, err := filepath.Abs(filename)
```

Comment:
It could be relative to the site root, but this code makes it relative to
the pwd.

---
https://onename.com/tgulacsi

2015-09-01 16:29 GMT+02:00 Matt Holt <notifications@github.com>:

> In middleware/basicauth/basicauth.go
> <https://github.com/mholt/caddy/pull/228#discussion_r38424614>:
>
> >  	Resources []string
> >  }
> > +
> > +type PasswordMatcher func(pw string) bool
> > +
> > +var (
> > +	htpasswords   map[string]map[string]PasswordMatcher
> > +	htpasswordsMu sync.Mutex
> > +)
> > +
> > +func GetHtpasswdMatcher(filename, username string) (PasswordMatcher, error) {
> > +	filename, err := filepath.Abs(filename)
>
> Before this, is the filename relative to the site root or the pwd? It
> seems to me it's from the pwd, and I'm wondering if it would make more
> sense to be relative to site root. What do you think?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mholt/caddy/pull/228/files#r38424614>.
>